src/values/instruction_value.rs

```diff
@@ -684,16 +688,42 @@ impl<'ctx> InstructionValue<'ctx> {
         }
 
         let is_basic_block = unsafe { !LLVMIsABasicBlock(operand).is_null() };
-
         if is_basic_block {
             let bb = unsafe { BasicBlock::new(LLVMValueAsBasicBlock(operand)) };
 
-            Some(Operand::Block(bb.expect("BasicBlock should always be valid")))
-        } else {
-            Some(Operand::Value(unsafe { BasicValueEnum::new(operand) }))
+            return Some(Operand::Block(bb.expect("BasicBlock should always be valid")));
+        }
+
+        match self.try_ingest_metadata(operand) {
+            // This is indeed metadata.
+            Ok(Some(metadata)) => return Some(Operand::Metadata(metadata)),
+            // This is null metadata, which we can't convert to an operand.
+            Err(_) => return None,
```

Comment:
This error should be passed on to the caller, which means that this function should return a Result.

Reply:
If not breaking compatibility results in an uncaught error, what choice is there? As it's written now, it could cause subtle bugs that would be hard to trace.

Reply:
Good point.